src/com/amazon/corretto/crypto/provider/ConstantTime.java

```diff
@@ -51,4 +53,53 @@ static final int select(int selector, int a, int b) {
 
     return b ^ (combined & mask);
   }
+
+  /**
+   * All the bytes in the trusted buffer between trustedStart and trustedStart + trustedLen are
+   * examined if trustedLen == otherLen. Please make sure the trusted array is the one that we
+   * trust.
+   *
+   * @param trusted
+   * @param trustedStart
+   * @param trustedLen
+   * @param other
+   * @param otherStart
+   * @param otherLen
+   * @return true iff all the bytes in the specified ranges are equal
+   */
+  static boolean equals(
+      final byte[] trusted,
+      final int trustedStart,
+      final int trustedLen,
+      final byte[] other,
+      final int otherStart,
+      final int otherLen) {
+
+    if (trustedLen != otherLen) {
+      return false;
+    }
+
+    int result = 0;
+
```

Comment:
Not entirely sure what you're definition of trust is, but we're extending a lot of it here. No size checking, no checking for negative indices... It would help to define exactly what we're trusting.

Reply:
In the first implementation, `equals(a, b)` would not necessarily take the same time as `equals(b, a)` since the the length of the first input would determine the number of iterations. In that implementation, one needed to ensure the first argument is something that has been vetted: a large untrusted buffer as first arg could potentially be problematic.
With the new implementation, the performance of `equals(a, b)` is exactly the same as `equals(b, a)` so the "trusted" concept is irrelevant now. I forgot to update the comment and will do it now.